### DISCUSSION TITLE
[openssl-linux] struggling to include

### DISCUSSION THREAD

**Discussion by Dariusz1989** [microsoft/vcpkg#23069] [Q&A]
Hey

Today is my 1st day with linux centos9 & trying to wrap my head around how can I get vcpkg to do openssl for me.

When I do find_package(OpenSSL REQUIRED) hes finding the OS one and not vcpkg.
If I try with > 

find_package(OpenSSL REQUIRED PATHS vcpkg/vcpkg/installed/x64-linux/share/)

I get 
  Could not find a package configuration file provided by "OpenSSL" with any
  of the following names:

    OpenSSLConfig.cmake
    openssl-config.cmake

Is this library damaged? How do I include it on linux properly? I'm a tad lost.
Thanks!

**Comment by dg0yt** [comment]
So do you use the vcpkg cmake toolchain?

**Reply by Dariusz1989** [reply]
I hope I do, I included the vcpkg cmake file. 
I cant use find_package(openSSL) as that return system openssl aso I have to explicitly pass correct path for vcpkg one. But sadly that does not work as I dont know what to pass o.o

**Reply by dg0yt** [reply]
Including the file is not the same as using it as a toolchain file (via `CMAKE_TOOLCHAIN_FILE`).  
And note the toolchain file must be set on the very first run of cmake for a working directory.

**Reply by Dariusz1989** [reply]
Hai @dg0yt 
Thanks for help!

Hmm so instead of 
    ```include(${VCPKG_ROOT}scripts/buildsystems/vcpkg.cmake)```
I ned to do
    ```set(CMAKE_TOOLCHAIN_FILE ${VCPKG_ROOT}scripts/buildsystems/vcpkg.cmake)```

?
```
And note the toolchain file must be set on the very first run of cmake for a working directory.
```
What does this mean, 
My Cmake includes another cmake that does most of my config, can I place it there? Or I have to place it in 1st cmake that run project?
```
cmake_minimum_required(VERSION 3.20)
project(testApps)
get_filename_component(testAppsRoot"${CMAKE_CURRENT_SOURCE_DIR}/../../../apps/" ABSOLUTE)
include(${testAppsRoot}/coreApp/coreCMAKE/CMakeLists.txt)
```
Can I place it inside > ```coreCMAKE/CMakeLists.txt``` Or it has to be above project(testApps)

I feel a little... "scared" of setting toolchain here, after brief reading, it seems like it configures a lot of things and I don't think I want it :c I just want to include what I need from vcpkg without it remastering my project :- (

**Reply by Neumann-A** [reply]
> What does this mean,
>  Or it has to be above project(testApps)

It has to be above the very first `project()` call otherwise it won't take effect.
